🚀 3단계 - 배포 스크립트 작성하기

90mansik opened the pull request:
안녕하세요 영훈님 😄 
3단계 미션 리뷰 요청 드립니다.

바쁘신 와중에도 신경 써주셔서 감사합니다.
덕분에 많이 배우는 과제가 되었습니다!!
정말 감사합니다 🙏

Commit:
feat: 배포 스크립트 생성

Commit:
fix: Step2 빌드 오류 설정 파일 수정

chae-yh commented on sh/subway_deploy.sh:
```diff
@@ -0,0 +1,84 @@
+ if [ -z $PID ]; then
+         echo -e ">> Process Not Exists"
+ else
+       echo -e  ">> Kill -9 $PID"
```
kill 옵션은 여러가지가 있는데 혹시 -9로 주신 이유가 있을까요...?

90mansik replied:
현업에서 정상적으로 작동하지 않는 프로세스를 죽일때 강제종료 옵션인 -9 을 써서 사용해왔는데요.
안전하게 종료하기 위해 -15 옵션을 주는 게 좋을 것 같네요!

chae-yh commented on sh/subway_deploy.sh:
```diff
@@ -0,0 +1,84 @@
+ echo -e ""
+ echo -e ">> Process ReStart 🏃♂️"
+   echo -e "nohup java -jar -Dspring.profiles.active=local  $LIBRAY_PATH/$APP_NAME 1> $PROJECT_PATH/service.log 2>&1  &"
+   nohup java -jar -Dspring.profiles.active=local $LIBRAY_PATH/$APP_NAME 1> $PROJECT_PATH/service.log 2>&1  &
```
active 를 local이 아니라 prod 로 주셔야하지 않을까요...? 사이트에 데이터가 노출이 안되네요..ㅠ

90mansik replied:
넵 수정하겠습니다!

chae-yh commented on src/main/resources/application-test.properties:
```diff
@@ -1,5 +1,13 @@
 
+spring.h2.console.enabled=true
+
+spring.datasource.driver-class-name=org.h2.Driver
```
테스트 DB를 h2로 설정 👍

chae-yh commented on sh/subway_deploy.sh:
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
```
현재 구조라면, 실제로 git에서 코드 변화가 있더라도 서비스에 반영은 안될 것 같아요. 해당 스크립트를 crontab 에 걸어서 주기적으로 실행해주면 어떨까요..?

90mansik replied:
넵 10분 주기로 호출하도록 conrtab 걸어 놓겠습니다!

chae-yh replied:
`*/10 * * * * /home/ubuntu/subway_deploy.sh > /home/ubuntu/crontab.log 2>&1
`
로그도 파일로 잘 남겨주셨네요 👍 다만 log에 넣을 때 > 이렇게 적용하셔서, 데이터가 크론탭이 수행될 때만 남는 것 같아요 >> 이렇게 실행될 때마다 파일의 뒤에 붙여주어도 좋을 것 같습니다 👍

chae-yh commented on sh/subway_deploy.sh:
```diff
@@ -0,0 +1,84 @@
+movePath
+
+# git pull
+pull
```
항상 pull을 하지말고, git remote에 변경이 있을 때만 동작하도록 하면 어떨까요...? 강의에도 나온 변경감지를 사용하면 될 것 같습니다..!

90mansik replied:
적용 했습니다!!
감사합니다

chae-yh commented on src/main/resources/application-test.properties:
```diff
@@ -1,5 +1,13 @@
+
+spring.datasource.driver-class-name=org.h2.Driver
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.username=sa
```
step2의 과제이기는 한데, 이렇게 설정파일에 계정 정보가 노출되면 안됩니다. 서브 모듈을 이용해서 숨겨주면 어떨까요...?

90mansik replied:
안녕하세요 리뷰어님!
서브 모듈에 대해서 공부해보고 적용하도록 하겠습니다!
제가 git도 처음이고 인프라도 처음이고 해서 시간이 좀 걸릴 것 같습니다!
우테코가 끝나는 시점에 적용해봐도 될까요?

chae-yh reviewed (changes requested):
지용님 안녕하세요. 벌써 마지막 step이네요. 그리고 step3도 전체적으로 환경 구성 잘해주셨습니다 👍 다만 바로 merge 하기에는 몇가지 고민해주시면 좋을만한 내용들이 있어 코멘트 드리니 확인 부탁드립니다.

마지막까지 파이팅 입니다 : )

Commit:
fix: Step3 피드백 반영

- kill 옵션 변경
AS IS : -9(강제종료)
TO BE : -15(정상종료)
- git diff check 기능 추가
- 프로세스 실행 옵션 변경
AS IS : spring.profiles.active=test
TO BE : spring.profiles.active=prod

chae-yh reviewed (approved):
지용님 안녕하세요. 피드백 잘 반영해주셨습니다 👍 끝까지 포기하지 않으시고, 미션을 끝내주시는 부분이 인상적이네요 ㅎㅎ 살짝 고민해주시면 좋을만한 내용이 있어 코멘트 드렸지만, 전체적으로 미션의 목적을 잘 달성해주셨다고 생각되어 merge 하겠습니다...!

그리고 여기까지가 제가 지용님께 드리는 마지막 리뷰인 것 같습니다. 쉽지 않은 미션이고, 피드백 많이 드렸었는데 잘 반영해주셔서 감사했습니다 ㅎㅎ 나름대로는 열심히 리뷰 드리려고 노력했는데, 그 마음이 지용님께 조금이라도 가 닿으면 좋겠네요...!
이제 코스도 거의 마지막을 향해 달려가고 있는데, 지금처럼 포기없이 꼭 완주하시길 빕니다..!

그 동안 고생 많으셨습니다 감사합니다 : )
